libevmasm/ConstantOptimiser.cpp

```diff
@@ -155,36 +158,72 @@ AssemblyItems CodeCopyMethod::execute(Assembly& _assembly) const
 	bytes data = toBigEndian(m_value);
 	assertThrow(data.size() == 32, OptimizerException, "Invalid number encoding.");
 	AssemblyItems actualCopyRoutine = copyRoutine();
-	actualCopyRoutine[4] = _assembly.newData(data);
+	if (m_params.evmVersion.hasPush0())
+		actualCopyRoutine[3] = _assembly.newData(data);
+	else
+		actualCopyRoutine[4] = _assembly.newData(data);
```

Comment:
I added a parameter to `copyRoutine` in this last [commit](https://github.com/ethereum/solidity/pull/14117/commits/202678743e7af16394dd69d5b58b04d83d006a27). Since  `AssemblyItem` is forward declared in `ConstantOptimiser.h`, we need to use a pointer.